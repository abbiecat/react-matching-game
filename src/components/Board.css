.Board-btn {
  -webkit-border-radius: 2;
  -moz-border-radius: 2;
  border-radius: 2px;
  font-family: Arial;
  color: #ffffff;
  font-size: 20px;
  background: #a1a1a1;
  padding: 10px 20px 10px 20px;
  text-decoration: none;
}

.Board-btn:hover {
  background: #b5b5b5;
  text-decoration: none;
}
