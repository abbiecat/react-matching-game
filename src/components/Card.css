.Card {
  height: 100px;
  width: 100px;
  background: aliceblue;
  margin: 10px;
  display: inline-block;
  border-radius: 5px;
  border: 3px solid #e1e9f1;
}

.Card:hover {
  background: #d2d8de;
}

.Card-icon {
  margin-top: 15px;
}