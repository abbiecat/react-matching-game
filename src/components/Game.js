import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as gameActions from '../actions/game';

import Board from './Board';

export class Game extends Component {
  render() {
    return(
      <Board selectCard={this.props.actions.selectCard} addGuess={this.props.actions.addGuess} restartGame={this.props.actions.restartGame} cards={this.props.cards} game={this.props.game}/>
    );
  }
}

function mapStateToProps(state, props) {
  return {
    cards: state.cards,
    game: state.game
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(Object.assign({}, gameActions), dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Game);
