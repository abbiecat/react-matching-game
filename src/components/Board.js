import React, { Component } from 'react';
import '../styles/Board.css';
import Card from './Card';

export class Board extends React.Component {
  renderCard(i) {
    const card = this.props.cards[i];
    return (
      <Card
        value={i}
        icon={card.icon}
        selected={card.selected}
        matched={card.matched}
        selectCard={this.props.selectCard}
        addGuess={this.props.addGuess}
      />
    );
  }

  render() {
    return (
      <div>
        <div>
          <div>
            {this.renderCard(0)}
            {this.renderCard(1)}
            {this.renderCard(2)}
            {this.renderCard(3)}
          </div>
          <div>
            {this.renderCard(4)}
            {this.renderCard(5)}
            {this.renderCard(6)}
            {this.renderCard(7)}
          </div>
          <div>
            {this.renderCard(8)}
            {this.renderCard(9)}
            {this.renderCard(10)}
            {this.renderCard(11)}
          </div>
          <div>
            {this.renderCard(12)}
            {this.renderCard(13)}
            {this.renderCard(14)}
            {this.renderCard(15)}
          </div>
        </div>
        <h2>Guesses: {Math.floor(this.props.game.guesses/2)}</h2>
        <h2>Matches: {this.props.game.matches}</h2>
        <button className={'Board-btn '} onClick={(e) => {this.props.restartGame()}}>Play Again</button>
      </div>
    );
  }
}

export default Board;
