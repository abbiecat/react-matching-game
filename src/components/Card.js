import React, {Component} from 'react';
import './Card.css'
import logo from '../logo.svg';
import FontAwesome from 'react-fontawesome';

class Card extends React.Component {
  render() {
    return (
      <div className='Card' onClick={() => this.props.selectCard('foo')}>
        <FontAwesome className={'Card-icon'} name={this.props.icon} size={'3x'} style={{ textShadow: '0 1px 0 rgba(0, 0, 0, 0.1)' }}/>
      </div>
    )
  }
}

export default Card
