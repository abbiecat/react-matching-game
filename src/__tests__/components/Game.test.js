import React from 'react';
import { shallow } from 'enzyme';

import { Game } from '../../components/Game';
import { Provider } from 'react-redux';

test('render Game', () => {
  const wrapper = shallow(
    <Game actions={{selectCard: jest.fn(), addGuess: jest.fn(), restartGame: jest.fn()}} cards={[{icon: 'flash', selected: false, matched: false}]} game={{}}/>
  );
  expect(wrapper).toMatchSnapshot();
});
