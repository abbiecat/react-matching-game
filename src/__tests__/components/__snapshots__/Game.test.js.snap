// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Game 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Game
    actions={
        Object {
            "addGuess": [Function],
            "restartGame": [Function],
            "selectCard": [Function],
          }
    }
    cards={
        Array [
            Object {
              "icon": "flash",
              "matched": false,
              "selected": false,
            },
          ]
    }
    game={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addGuess": [Function],
      "cards": Array [
        Object {
          "icon": "flash",
          "matched": false,
          "selected": false,
        },
      ],
      "game": Object {},
      "restartGame": [Function],
      "selectCard": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addGuess": [Function],
        "cards": Array [
          Object {
            "icon": "flash",
            "matched": false,
            "selected": false,
          },
        ],
        "game": Object {},
        "restartGame": [Function],
        "selectCard": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
